pennylane/ops/qubit/observables.py

```diff
@@ -235,6 +237,15 @@ def label(self, decimals=None, base_label=None):
         basis_string = "".join(str(int(i)) for i in self.parameters[0])
         return f"|{basis_string}⟩⟨{basis_string}|"
 
+    # pylint:disable=unused-argument
+    @staticmethod
+    def compute_matrix(*params, **hyperparams):
+        basis_state = params[0]
+        m = np.zeros((2 ** len(basis_state), 2 ** len(basis_state)))
+        idx = int("".join(str(i) for i in basis_state), 2)
+        m[idx, idx] = 1
+        return m
```

Comment:
I suppose the casting happens later now?

Reply:
Oh good point! I added this because I saw it's lacking and easy to implement (famous last words). 

However I doubt here we can infer the framework, because `basis_state` is usually a list?

Reply:
Oh so this is really interesting - `Projector` accepts `basis_state` as `tensor_like`, but in `__init__` it casts it first to a numpy array, and then to a list.

It really shouldn't be doing this, it should be leaving it as a tensor

Reply:
Technically as well, `basis_state` is not a parameter but a hyperparameter, since it is not trainable?

Reply:
I think I am to blame, I added this operator iirc!